common/src/test/scala/uk/ac/wellcome/sqs/SQSReaderTest.scala

```diff
@@ -15,44 +15,81 @@ class SQSReaderTest
     with IntegrationPatience
     with SQSLocal {
 
+  override def queueName: String = "id_minter_queue"
+
   it("should get messages from the SQS queue, limited by the maximum number of messages and return them") {
     val sqsConfig =
-      SQSConfig("eu-west-1",
-                idMinterQueueUrl,
-                waitTime = 20 seconds,
-                maxMessages = 2)
+      SQSConfig("eu-west-1", queueUrl, waitTime = 20 seconds, maxMessages = 2)
     val messageStrings = List("someMessage1", "someMessage2", "someMessage3")
-    messageStrings.foreach(sqsClient.sendMessage(idMinterQueueUrl, _))
+    messageStrings.foreach(sqsClient.sendMessage(queueUrl, _))
     val sqsReader =
       new SQSReader(sqsClient, sqsConfig)
 
-    val futureMessages = sqsReader.retrieveMessages()
+    var receivedMessages: List[Message] = Nil
 
-    whenReady(futureMessages) { messages =>
+    val futureMessages = sqsReader.retrieveAndDeleteMessages(message => receivedMessages = message :: receivedMessages)
+
+    whenReady(futureMessages) { _ =>
       // SQS is not a FIFO queue and it only guarantees that a message is sent at least once,
       // not that it is received exactly once
-      messages should have size 2
-      messages.foreach { message =>
+      receivedMessages should have size 2
+      receivedMessages.foreach { message =>
         messageStrings should contain(message.getBody)
       }
     }
+
+    // Check that the previous 2 messages have been deleted
+    assertNumberOfMessagesAfterVisibilityTimeoutIs(1, sqsReader)
   }
 
-  it("should return a failed future if writing to the SNS topic fails") {
+  it("should return a failed future if reading from the SQS queue fails") {
     val sqsConfig =
       SQSConfig("eu-west-1",
                 "not a valid queue url",
                 waitTime = 20 seconds,
                 maxMessages = 1)
     val sqsReader = new SQSReader(sqsClient, sqsConfig)
 
-    val futureMessages = sqsReader.retrieveMessages()
+    val futureMessages = sqsReader.retrieveAndDeleteMessages(_ => ())
 
     whenReady(futureMessages.failed) { exception =>
       exception.getMessage should not be (empty)
     }
   }
 
-  private def createMessage(jsonMessage: String) =
-    new Message().withBody(jsonMessage)
+  it("should return a failed future if processing one of the messages fails - the failed message should not be deleted") {
+    val sqsConfig =
+      SQSConfig("eu-west-1", queueUrl, waitTime = 20 seconds, maxMessages = 10)
+
+    val failingMessage = "failingMessage"
```

Comment:
Can we pick a string that’s different from the variable names? Makes logs easier to debug, because you can tell whether you’re looking at a traceback or an output. For example:

```scala
val failingMessage = "This message will fail"
val messageStrings = List("This is the first message", failingMessage, "This is the final message")
```